server/service/src/document/update_trigger.rs

```diff
@@ -0,0 +1,106 @@
+use std::str::FromStr;
+
+use chrono::NaiveDate;
+use repository::{
+    Document, Gender as NameRowGender, NameRow, NameRowRepository, NameStoreJoinRepository,
+    NameStoreJoinRow, NameType, StorageConnection,
+};
+
+extern crate schemafy_core;
+extern crate serde;
+extern crate serde_json;
+
+use serde::{Deserialize, Serialize};
+use util::uuid::uuid;
+
+use super::document_service::DocumentInsertError;
+
+schemafy::schemafy!("src/document/schemas/patient.json");
+
+/// Callback called when the document has been updated
+pub fn document_updated(
+    con: &StorageConnection,
+    store_id: &str,
+    doc: &Document,
+) -> Result<(), DocumentInsertError> {
+    match doc.r#type.as_str() {
+        "Patient" => {
+            let patient: Patient = serde_json::from_value(doc.data.clone()).map_err(|err| {
```

Comment:
I don't think we would be able to do this dynamically (or is this just an example of callback ?)

Reply:
This is only for the well known type right now, i.e. where you know the expected type

Reply:
But couldn't you have a custom form/schema for patient entry ?

Reply:
You've mentioned 'deleted' fields in reply to one of my comments in another PR, is that going to be ok for this ?

Reply:
Yes, what I meant is schema where there is an optional type and a user unsets a previous set field, e.g. `data.field = undefined` so the schema is still satisfied.